### PR TITLE
Add asix-mcs783x 1.1.0 depends_on

### PR DESCRIPTION
**Pull request by ytbmulder**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

### CODE REVIEW

**Comment by ytbmulder** [comment]
The CI fails due to the added `depends_on` stanza. 
The vendor states only compatibility until `yosemite`. 
I installed and uninstalled this cask on `sierra` without a problem.

**Comment by ytbmulder** [comment]
@vitorgalvao The CI ran into a timeout during the download of the installer.

**Comment by vitorgalvao** [comment]
> uninstall script uninstal driver does not exist.

File: Casks/asix-mcs783x.rb [hunk 1]
```diff
@@ -12,11 +12,17 @@ def self.basename(version)
...
 
+  depends_on macos: '<= :yosemite'
   container nested: "#{Utils.basename(version)}/MCS7830_v#{version.major_minor_patch}.dmg"
 
   pkg "MCS7830 v#{version.major_minor_patch}.pkg"
 
-  # The "uninstal" (one "l") isn't a typo, that's the exact filename
-  uninstall script:  { executable: 'uninstal driver' },
+  postflight do
+    system_command '/sbin/kextload',
```
Comment (reitermarkus) [inline Casks/asix-mcs783x.rb]: Why does it have to be loaded manually?

**Comment by reitermarkus** [comment]
>> uninstall script uninstal driver does not exist.

Obviously, since the install fails due to the `depends_on :macos`, so this error can be ignored.

File: Casks/asix-mcs783x.rb (hunk 1 above)
Comment (ytbmulder) [reply Casks/asix-mcs783x.rb]: After installation, the installed `kext` is not loaded, similar to the `asix-ax88179.rb` cask from the same developer. See below:

https://github.com/Homebrew/homebrew-cask-drivers/blob/9e2bd43b1c7f1aaafd9169a9c9af630b18b8e335/Casks/asix-ax88179.rb#L13-L17

In my opinion, both casks should share a common installation procedure. 
Either both load the installed `kext`, or both do not.

**Pull request merged**